### DISCUSSION TITLE
JWT authorization : who check if the token is still valid ?

### DISCUSSION THREAD

**Discussion by chddv** [bluenviron/mediamtx#3814] [Questions]
### Question

Hi,
I think I missed something with the JWT authentication mechanism.
I have understood  that the client need to ask an identity server for a JWT.
This token is received by the client and send back to mediamtx. Mediamtx only check the content of the token to get the mediamtx authorization.
After that, who is going to check if the token is still valid ? (for date time ...) 
I'm trying to implement some kind of "magic url"  to acces to mediamtx from outside of a network.
But I still bother my self arround this JWT validation.
Peraps I should simply left the jwt mechanism , and simply used http auth.
Thanks.
Christophe

**Comment by aler9** [comment]
Hello, you are right, the JWT is checked during handshake only but it's not checked during streaming. I've created a feature request in order to track the issue: #4300
